src/sentry/relay/config/experimental.py

```diff
@@ -60,11 +60,32 @@ def add_experimental_config(
     NOTE: Only use this function if you expect Relay to behave reasonably
     if ``key`` is missing from the config.
     """
+
+    if subconfig := build_safe_config(key, function, *args, **kwargs):
+        config[key] = subconfig
+
+
+R = TypeVar("R")
+R_default = TypeVar("R_default")
+
+
+def build_safe_config(
+    key: str,
+    function: Callable[..., R],
+    *args: Any,
+    default_return: R_default | None = None,
+    **kwargs: Any,
```

Comment:
this should be `Callabe[P, R]` and `*args: P.args` and `**kwargs: P.kwargs` (`P` is the conventional name for `ParamSpec`) such that the signature is preserved in callers

Reply:
I did this in a first place, but maybe I did something wrong since I had some problems with mypy because I am inserting extra parameter (`timeout`) when calling a function (line 88 in this PR):

This was an error:
```
src/sentry/relay/config/experimental.py:88: error: Argument 1 has incompatible type "TimeChecker"; expected "P.args"  [arg-type]
```

Reply:
is this file in the ignore list?  I suspect you wrote `*args: *P.args` which mypy should be flagging separately as a mistake